src/coll/mod.rs

```diff
@@ -246,6 +253,68 @@ impl<T> Collection<T> {
         self.drop_common(options, session).await
     }
 
+    #[cfg(feature = "csfle")]
+    #[allow(clippy::needless_option_as_deref)]
+    async fn drop_aux_collections(
+        &self,
+        options: Option<&DropCollectionOptions>,
+        mut session: Option<&mut ClientSession>,
+    ) -> Result<()> {
+        // Find associated `encrypted_fields`:
+        // * from options to this call
+        let mut enc_fields = options.and_then(|o| o.encrypted_fields.as_ref());
+        let enc_opts = self.client().auto_encryption_opts().await;
+        // * from client-wide `encrypted_fields_map`:
+        let client_enc_fields = enc_opts
+            .as_ref()
+            .and_then(|eo| eo.encrypted_fields_map.as_ref());
+        if enc_fields.is_none() {
+            enc_fields =
+                client_enc_fields.and_then(|efm| efm.get(&format!("{}", self.namespace())));
+        }
+        // * from a `list_collections` call:
+        let found;
+        if enc_fields.is_none() && client_enc_fields.is_some() {
+            let filter = doc! { "name": self.name() };
+            let specs: Vec<_> = match session.as_deref_mut() {
+                Some(s) => {
+                    let mut cursor = self
+                        .inner
+                        .db
+                        .list_collections_with_session(filter, None, s)
+                        .await?;
+                    cursor.stream(s).try_collect().await?
+                }
+                None => {
+                    self.inner
+                        .db
+                        .list_collections(filter, None)
+                        .await?
+                        .try_collect()
+                        .await?
+                }
+            };
+            for spec in specs {
```

Comment:
since we pass a filter of `{ "name": self.name() }` which is an exact match, this cursor will only ever contain a single result or none. I think it might be easier to grok if we rewrite this for loop as an `if let` instead to better reflect that.

Reply:
Done.